### DISCUSSION TITLE
USB Connection to ESP32

### DISCUSSION THREAD

**Discussion by F4ight** [StuckAtPrototype/Racer#30] [Ideas]
Hello,

Cool project!  I am in the process of getting quotes from JLCPCB and PCBWay to build 5 kits but was wondering if there was a reason why pins 26 and 27 were not connected to the USB connector so it could be communicated with directly over USB or through something like a CP2102.

Thank you

**Comment by DapperDan81** [comment]
Hi,

do you mean this?
https://github.com/StuckAtPrototype/Racer/issues/8#issue-2595239563

**Reply by F4ight** [reply]
Ha, yes.  Exactly like that comment.

**Comment by markus7h** [comment]
> Cool project!  I am in the process of getting quotes from JLCPCB and PCBWay to build 5 kits 
...

@F4ight have You been successful with JLC and if so, would you mind to share the pick&place file. Because this is what is hindering me to order.

Thank You 
Markus

**Reply by F4ight** [reply]
PCBWay wanted way too much for assembly ($88 per board) and JLC charges $3 per part for extended parts per board so I just ordered the PCBs and stencils from JLC.  10 pieces of each of the three boards with blue masks was $43 but shipping was $52 which is lame for something as light as a PCB.

Attached are the three component placement files that I exported from KiCAD.  Hopefully you have better luck with a fab shop.

[controller-top-pos.csv](https://github.com/user-attachments/files/17594317/controller-top-pos.csv)
[sensors-top-pos.csv](https://github.com/user-attachments/files/17594318/sensors-top-pos.csv)
[racer-top-pos.csv](https://github.com/user-attachments/files/17594319/racer-top-pos.csv)
